### PR TITLE
changed loops for lambda

### PR DESCRIPTION
**Pull request by MaVaYuran**

### CODE REVIEW

File: src/main/java/by/mariayuran/Order.java [hunk 1]
```diff
@@ -73,19 +74,21 @@ public void makeOrderCompleted() {
...
+
     public void addBookToOrder(List<Book> books) {
-      Scanner  scanner = new Scanner(System.in);
+        Scanner scanner = new Scanner(System.in);
         int orderSize = scanner.nextInt();
-        for (int i = 0; i < orderSize; i++) {
-            this.addBook(getAnyBook(books));
-        }
+        IntStream.range(0, orderSize)
+                .forEach(i -> this.addBook(getAnyBook(books)));
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/Order.java:82]: what is i?

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@MaVaYuran 
Hi) please, fix the comment. and add logic to work with files

**Comment by Alexmansar** [comment]
@MaVaYuran I add review, why you ask again without any updates?

File: src/main/java/by/mariayuran/Order.java (hunk 1 above)
Comment (MaVaYuran) [reply src/main/java/by/mariayuran/Order.java:82]: this is method for entering number of books in order from console

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@MaVaYuran Hi)) good job)) please, add tests fr new methods as well))

File: src/main/java/by/mariayuran/Order.java (hunk 1 above)
Comment (Alexmansar) [reply src/main/java/by/mariayuran/Order.java]: i. use name

File: src/main/java/by/mariayuran/Main.java [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+import by.mariayuran.library.LibraryRepositoryImpl;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 public class Main {
     public static void main(String[] args) {
-        BookStore bookStore = new BookStore();
-        ObjectMapper objectMapper = null;
+         LibraryRepository libraryRepository = new LibraryRepositoryImpl("library.json");
+        BookStore bookStore = new BookStore(libraryRepository);
+//        ObjectMapper objectMapper = null;
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/Main.java:11]: remove please

File: src/main/java/by/mariayuran/Order.java [hunk 3]
```diff
@@ -113,22 +114,22 @@ public void addBookToOrder(List<Book> lib) {
...
+//        objectMapper = new ObjectMapper();
+//        objectMapper.registerModule(new JavaTimeModule());
+//        try {
+//            objectMapper.writeValue(new File(filePath), orders);
+//            System.out.println("Orders were written to " + filePath);
+//        } catch (IOException e) {
+//            System.out.println("Orders weren't written to json");
+//            e.printStackTrace();
+//        }
+//    }
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/Order.java:132]: and here

File: src/test/java/by/mariayuran/OrderTest.java [hunk 4]
```diff
@@ -1,33 +1,34 @@
...
+    @Mock
+    LibraryRepository libraryRepository;
+
+
 
 
     @BeforeEach
     public void setUp() {
-        bookStore = new BookStore();
+//        bookStore = new BookStore(libraryRepository);
```
Comment (Alexmansar) [inline src/test/java/by/mariayuran/OrderTest.java:31]: same

File: src/main/java/by/mariayuran/library/LibraryRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            return objectMapper.readValue(inputStream, new TypeReference<List<Book>>() {
+            });
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to load library from " + filePath, e);
+        }
+    }
+
+    @Override
+    public void writeOrderToJson(List<Order> orders) {
+        String filePath = "src/main/resources/orders.json";
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/library/LibraryRepositoryImpl.java:42]: In lines 29 and 42, we use different approaches. Can we use the same one?

**Review by Alexmansar — CHANGES_REQUESTED** [review]

File: src/main/java/by/mariayuran/library/LibraryRepositoryImpl.java (hunk 5 above)
Comment (MaVaYuran) [reply src/main/java/by/mariayuran/library/LibraryRepositoryImpl.java:42]: not sure, because there are different filepath

Comment (Alexmansar) [reply src/main/java/by/mariayuran/library/LibraryRepositoryImpl.java:42]: try (InputStream inputStream = getClass().getClassLoader().getResourceAsStream(filePath)) {
can we use in  writeOrderToJson?

**Pull request merged**